stdlib/public/stubs/UnicodeNormalization.cpp

```diff
@@ -36,6 +36,11 @@
 
 #pragma clang diagnostic pop
 
+// clang 4.0 under a recent linux has unicode with built-in char16_t support
+#if (defined(__CYGWIN__) || defined( _MSC_VER)) || defined(__linux__)
+#define __USE_UCHAR 1
```

Comment:
We should not be defining underscored macros like this.  Could you just add `__linux__` to the guards instead?